repository.go

```diff
@@ -8,122 +8,159 @@ import (
 	_ "github.com/lib/pq"
 )
 
-const (
-	host     = "localhost"
-	port     = 5432
-	user     = "postgres"
-	password = "1234"
-	dbname   = "test"
-)
-
-func getCart() {
-
-	psqlInfo := fmt.Sprintf("host=%s port=%d user=%s "+
-		"password=%s dbname=%s sslmode=disable",
-		host, port, user, password, dbname)
-	db, err := sql.Open("postgres", psqlInfo)
-	if err != nil {
-		panic(err)
-	}
-	defer db.Close()
+func getItem(db *sql.DB) {
+	sqlStatement :=
+		"select item_id, item_name, item_value " +
+			"from public.item " +
+			"where item_id = ?; "
+	sqlStatement = strings.Replace(sqlStatement, "?", "1", 1)
+	fmt.Println(sqlStatement)
 
-	err = db.Ping()
-	if err != nil {
+	var item Item
+	row := db.QueryRow(sqlStatement)
+	switch err := row.Scan(&item.id, &item.name, &item.value); err {
+	case sql.ErrNoRows:
+		fmt.Println("No rows were returned!")
+	case nil:
+		fmt.Println(item)
+	default:
 		panic(err)
 	}
-	fmt.Println("Successfully connected!")
+}
 
+func getAllCart(db *sql.DB) {
 	sqlStatement :=
-		"select cart_id, user_name " +
-			"from public.cart " +
-			"where cart_id = ?; "
+		"SELECT cart.cart_id, cart.user_name, cart_items.item_amount, item.item_id, item.item_name, item.item_value " +
+			"FROM cart_items, cart, item " +
+			"WHERE cart.cart_id = ? AND cart_items.cart_id = cart.cart_id AND item.item_id = cart_items.item_id; "
 	sqlStatement = strings.Replace(sqlStatement, "?", "1", 1)
 	fmt.Println(sqlStatement)
 
+	var item Item
 	var cart Cart
 	row := db.QueryRow(sqlStatement)
-	switch err := row.Scan(&cart.id, &cart.UserName); err {
+	switch err := row.Scan(&cart.id, &cart.UserName, &item.id, &item.name, &item.value); err {
 	case sql.ErrNoRows:
 		fmt.Println("No rows were returned!")
 	case nil:
+		cart.Products = append(cart.Products, item)
 		fmt.Println(cart)
 	default:
 		panic(err)
 	}
 
 }
 
-func getItem() {
+func listDbItems(db *sql.DB) ([]Item, error) {
+
+	var specificError error
+	sqlStatement := fmt.Sprintf(`
+		select item_id, item_name, item_value 
+		from public.item`)
+
+	fmt.Println(sqlStatement)
 
-	psqlInfo := fmt.Sprintf("host=%s port=%d user=%s "+
-		"password=%s dbname=%s sslmode=disable",
-		host, port, user, password, dbname)
-	db, err := sql.Open("postgres", psqlInfo)
+	var item []Item
+	row, err := db.Query(sqlStatement)
 	if err != nil {
-		panic(err)
+		fmt.Println(err)
+		specificError = err
```

Comment:
You could return just here, and this way you would get rid of the `else` this is called early return, and we avoid growing the code horizontally but instead vertically